### PR TITLE
Track where the Payment Request API is available

### PR DESCRIPTION
**Pull request by joelochlann**
## Why are you going this?
We'd like to know how many visitors to the site could make use of the Payment Request API if we integrated it

The Payment Request API allows the browser to store & retrieve card details, resulting in a much slicker checkout experience, especially on mobile. But browser support is still limited and there are a few other caveats to its availability. See here for more info:  https://stripe.com/docs/payment-request-api

## How?
We created a new custom dimension in GA (`paymentRequestApiStatus`, number `21`) and we populate with one of the following values: 

* 'PaymentRequestAPINotAvailable': the PaymentRequest is not defined. 
* 'CanMakePaymentNotAvailable': the PaymentRequest's CanMakePaymentNotAvailable function is not availble.
* 'AvailableNotInUse: The payment API is available but the user is not using it.
* 'AvailableInUse': The payment API is available and the user is using it.
* 'PaymentRequestAPIError': There was an exception in the block which is testing the payment request API.

### CODE REVIEW

File: assets/helpers/paymentIntegrations/stripeCheckout.js [hunk 1]
```diff
@@ -82,8 +108,3 @@ const openDialogBox = (amount: number, email: string) => {
     });
   }
 };
-
-export {
```
Comment (svillafe) [inline assets/helpers/paymentIntegrations/stripeCheckout.js]: We tend to prefer to export an object at the bottom of the file over having exports along the file.

File: assets/helpers/paymentIntegrations/stripeCheckout.js [hunk 2]
```diff
@@ -46,6 +46,14 @@ const getStripeKey = (currency: string, isTestUser: boolean) => {
   return stripeKey;
 };
 
+export const getStripeAccountCountry = (currency: string): string => {
```
Comment (svillafe) [inline assets/helpers/paymentIntegrations/stripeCheckout.js]: Try to use Currency types

File: flow-typed/global.js [hunk 3]
```diff
@@ -0,0 +1 @@
+declare var PaymentRequest: any;
```
Comment (Ap0c) [inline flow-typed/global.js]: Does flow throw errors without this?

Comment (svillafe) [reply flow-typed/global.js]: Yes, "Could not resolve name PaymentRequest"

Comment (svillafe) [reply flow-typed/global.js]: I avoid that by using `window.PaymentRequest` ec2788e

**Comment by joelochlann** [comment]
👍 I can't actually approve because I created the PR...

File: assets/helpers/tracking/googleTagManager.js [hunk 4]
```diff
@@ -23,6 +26,58 @@ function getDataValue(name, generator) {
...
+        },
+      ];
+
+      const details = {
+        total: {
+          label: 'tracking',
+          amount:
+            {
+              value: '1',
+              currency: 'GBP',
```
Comment (rupertbates) [inline assets/helpers/tracking/googleTagManager.js]: Does specifying GBP have any impact on the ability to make a payment if the user is in the US for example?

Comment (svillafe) [reply assets/helpers/tracking/googleTagManager.js]: Good questions, I don't think so (I tested the feature by putting a dummy credit card and it detected correctly). Nevertheless, I switched to use the `getCurrency` method in badfba3

**Pull request merged**